src/v/storage/compaction_reducers.cc

```diff
@@ -234,7 +234,7 @@ copy_data_segment_reducer::filter(model::record_batch&& batch) {
 }
 
 ss::future<ss::stop_iteration> copy_data_segment_reducer::do_compaction(
-  model::compression original, model::record_batch&& b) {
+  model::compression original, model::record_batch b) {
```

Comment:
this can be squashed into previous commits that switch methods to coroutines.

Reply:
I have squashed.